src/secret_sharing/xor.rs

```diff
@@ -20,3 +26,155 @@ impl XorReplicated {
         self.right
     }
 }
+
+/// XOR secret share.
+///
+/// Bits are stored in the Little-Endian format. Accessing the first element
+/// like `s[0]` will return the LSB.
+///
+/// ## Example
+/// ```
+/// use raw_ipa::secret_sharing::XorSecretShare;
+///
+/// let s = XorSecretShare::<64>::from(2_u128);
+/// let b0 = s[0];
+/// let b1 = s[1];
+///
+/// assert_eq!(b0, 0);
+/// assert_eq!(b1, 1);
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct XorSecretShare<const N: usize>(BitVec<u8, Lsb0>);
```

Comment:
I think you want `BitArr` here. I believe you can also omit `Lsb0` because that's the default.

Reply:
This is [`BitArray`](https://docs.rs/bitvec/latest/bitvec/array/struct.BitArray.html), but that has a drawback in that const generic expressions aren't available yet.  I don't know if we'll hit that, but we should try to avoid the heap allocation for these so that we can keep this lightweight.  (We're explicitly not `Copy` on these, but it would be good if we can approximate it.)

Reply:
I am confused, I thought https://docs.rs/bitvec/latest/bitvec/vec/struct.BitVec.html is heap-allocated?

Reply:
I'll change it to `BitArray`. I wanted to explicitly specify `Lsb0` so that readers are aware of this constraint.